File: Algorithms/Math/PalindromeNumber.js
const isPalindrome = function (x) {
  if (x < 0) {
    return false;
  }
  x = String(x);
  let left = 0;
  let right = x.length - 1;
  while (left < right) {
    if (x[left] !== x[right]) {
      return false;
    }
    left++;
    right--;
  }

  return true;
};
console.log(isPalindrome(10));
console.log(isPalindrome(121));
console.log(isPalindrome(-121));

File: Algorithms/Math/Palindrome Number/PalindromeNumber.py
def is_palindrome(x):
    if x < 0:
        return False
    x = str(x)
    left = 0
    right = len(x) - 1
    while left < right:
        if x[left] != x[right]:
            return False
        left += 1
        right -= 1
    return True

# Test cases
print(is_palindrome(10))   
print(is_palindrome(121))   
print(is_palindrome(-121)) 